### DISCUSSION TITLE
중고거래 내역테이블

### DISCUSSION THREAD

**Discussion by cho-ja-young** [MajorIn-ABO/MajorIn_FE#34] [Q&A]
## 중고거래 내역테이블

- 중고거래내역id
- 중고거래id ( 외래키
- 산사람id
- 판사람id
- 거래성사날짜
- 거래성사여부

위의 컬럼으로 생각중인데, 고민되는 부분이 있어서!

1. 거래 성사 여부 컬럼

거래가 성사가 되었으니까 내역 테이블에 등록된게 아닐까? 란 생각이 들어서
그냥 거래 성사 여부를 중고거래테이블에서 지우지말고 그대로 놔두고 내역테이블에는 날짜만 등록되도록
하는게 어떤지

2. 거래자 작성

중고거래 테이블에서 판 사람을 저장해두는 컬럼이 있는데
중고거래내역테이블에 판사람과 산사람 둘 다 저장하는거 어떻게 생각하는지?

**Comment by JangGusWjd** [comment]
중고거래 내역 테이블은 거래를 완료한 경우인거지?
1. 나는 중고거래 테이블에는 그대로 놔두고 내역테이블에 날짜 등록하는게 좋을거같아. 왜냐면 프론트 입장에서 보면 중고거래 페이지에 판매중/판매완료로 구분하여 볼 수 있는데 거래가 완료된 것을 지워버리면 판매완료된 내역을 사용자는 볼 수 없고, 책들이 평균적으로 어느정도 가격에 거래가 이루어졌는지 볼 수 없으니까. 그러고 내역테이블에 날짜가 등록된 것을 보고 판매 여부를 확인하는데도 도움이 될 것 같고!

2. 판사람은 무조건 저장해야 될 것 같고, 산사람은 없는것도 괜찮을 것 같아. 왜냐하면 판사람은 판매 교재를 등록할 때 아이디를 쉽게 얻을 수 있는데, 산사람의 아이디는 얻기 힘들거야. 판매자가 거래가 완료되면 판매완료를 누르는데, 이렇게 되면 산사람의 아이디를 얻을 수 있는 방법이 어렵다고 생각하거든

**Reply by cho-ja-young** [reply]
오 일단 1번은 나랑 같은 생각이어서 그러면 이렇게 진행할게!

2번은 아 뭔가 아이디를 얻는 방법이 없나.... 원래는 우리 채팅으로 하는거라 확실히 아이디 얻는 방법이 좀 까다롭긴한데 ㅠ
그럼 일단은 없는 걸로 해두겠음!!
